File: bst/Song.cpp
#include "Song.h"
#include <iostream>

// Constructor to initialize member variables
Song::Song() : id(0), popularity(0), year(0), key(0), mode(0), duration_ms(0), time_signature(0),
               danceability(0.0), energy(0.0), loudness(0.0), speechiness(0.0), acousticness(0.0),
               instrumentalness(0.0), liveness(0.0), valence(0.0), tempo(0.0) {}

// Method to print song details
void Song::printSong() const {
    std::cout << "ID: " << id << ", Artist: " << artist_name << ", Track: " << track_name
              << ", Popularity: " << popularity << ", Year: " << year << ", Genre: " << genre
              << ", Tempo: " << tempo << " BPM, Duration: " << duration_ms / 1000.0 << " seconds"
              << ", Time Signature: " << time_signature << std::endl;
}

File: popularidad/song.py
# song.py
class Song:
    def __init__(self, song_id, artist_name, track_name, track_id, popularity, year, genre, danceability, energy,
                 key, loudness, mode, speechiness, acousticness, instrumentalness, liveness, valence, tempo,
                 duration_ms, time_signature):
        self.song_id = int(song_id)
        self.artist_name = artist_name
        self.track_name = track_name
        self.track_id = track_id
        self.popularity = int(popularity)
        self.year = int(year)
        self.genre = genre
        self.danceability = float(danceability)
        self.energy = float(energy)
        self.key = int(key)
        self.loudness = float(loudness)
        self.mode = int(mode)
        self.speechiness = float(speechiness)
        self.acousticness = float(acousticness)
        self.instrumentalness = float(instrumentalness)
        self.liveness = float(liveness)
        self.valence = float(valence)
        self.tempo = float(tempo)
        self.duration_ms = int(duration_ms)
        self.time_signature = int(time_signature)

    def __repr__(self):
        return f"Song(ID={self.song_id}, Track={self.track_name}, Artist={self.artist_name})"